src/Yesod/Page/RenderedRoute.hs

```diff
@@ -68,7 +81,29 @@ getRenderedRoute = do
   let (path, _query) = renderRoute route
   query <- reqGetParams <$> getRequest
 
-  pure $ RenderedRoute { renderedRoutePath = path, renderedRouteQuery = query }
+  pure $ RenderedRoute
+    { renderedApproot = ""
+    , renderedRoutePath = path
+    , renderedRouteQuery = query
+    }
+
+getRenderedRouteAbsolute
+  :: (MonadHandler m, Yesod (HandlerSite m), RenderRoute (HandlerSite m))
+  => m RenderedRoute
+getRenderedRouteAbsolute = do
+  route <- maybe (throwString "no route") pure =<< getCurrentRoute
+
+  -- When I just use _query, it's always empty. Why would renderRoute return
+  -- this tuple if the Route value (apparently) never has the query information?
+  let (path, _query) = renderRoute route
+  query <- reqGetParams <$> getRequest
```

Comment:
This seems suspect. Is this debt within `yesod`? Was the route config file originally suppose to include query params?

Are there tests to validate this is working properly?

Reply:
This is copied code from the non-absolute path. Indeed I found it confusing at the time, hence the comment. All of our existing tests are effectively exercising it as correct though.